File: array.py
from multiprocessing.sharedctypes import Value


class Queuearray:
 
   
    def __init__(self, size=100):
        self.q = [None] * size      
        self.capacity = size        
        self.front = 0              
        self.rear = -1              
        self.count = 0              
 
    
    def dequeue(self):
        
        if self.isEmpty():
            print(' Underflow')
            exit(-1)
        Value = self.q[self.front]
        print('delete', Value)
        self.front = (self.front + 1) % self.capacity
        self.count = self.count - 1
        return Value
 
    
    def enqueue(self, data):
       
        if self.isFull():
            print('Overflow')
            exit(-1)
        print('Insertion', data)
        self.rear = (self.rear + 1) % self.capacity
        self.q[self.rear] = data
        self.count = self.count + 1
 
    def peek(self):
        if self.isEmpty():
            print(' UnderFlow')
            exit(-1)
        return self.q[self.front]
 
    
    def size(self):
        return self.count
 
   
    def isEmpty(self):
        return self.size() == 0
 
   
    def isFull(self):
        return self.size() == self.capacity
 
 
if __name__ == '__main__':
 
    q = Queuearray(5)
 
    q.enqueue(1)
    q.enqueue(2)
    q.enqueue(3)
 
    print(' size is', q.size())
    print('The  top element is', q.peek())
    q.dequeue()
    print('The top element is', q.peek())
 
    q.dequeue()
    q.dequeue()
 
    if q.isEmpty():
        print('null')
    else:
        print('elements')

File: array.cpp
#include <iostream>

using namespace std;

int que[50];
int y= 50;
int front = - 1;
int rear = - 1;
void insert() {
int x;
if (rear == y - 1)
cout<<" Overflow"<<endl;
else {
front = 0;
cout<<" insert value  : "<<endl;
cin>>x;
rear++;
que[rear] = x;
}
}
void delete1() {
if (front == - 1) {
cout<<"Underflow";
} else {
cout<<"Element deleted  : "<< que[front] <<endl;
front++;;
}
}
void display () {
if (front == - 1 )
cout<<" empty"<<endl;
else {
cout<<" elements are : ";
for (int i = front; i <= rear; i++)
cout<<que[i]<<" ";
cout<<endl;
}
}
int main() {
int option;
cout<<"1) insert "<<endl;
cout<<"2) delete"<<endl;
cout<<"3) display "<<endl;
cout<<"4) Exit"<<endl;
do {
cout<<" choice : "<<endl;
cin>>option;
switch (option) {
case 1: insert();
break;
case 2: delete1();
break;
case 3: display ();
break;
case 4: cout<<"Exit"<<endl;
break;
default: cout<<"not found"<<endl;
break;
}
} while(option!=0);
return 0;
}
